modules/apps/wiki/wiki-web/src/main/resources/META-INF/resources/wiki/edit_page_attachment.jsp

```diff
@@ -104,13 +97,9 @@ Ticket ticket = TicketLocalServiceUtil.addTicket(user.getCompanyId(), User.class
 			if (searchContainer) {
 				var rowColumns = [];
 
-				var deleteURL = Liferay.PortletURL.createURL('<%= deleteURL.toString() %>');
-
-				deleteURL.setParameter('fileName', event.name);
-
 				rowColumns.push(event.name);
 				rowColumns.push(uploader.formatStorage(event.size));
-				rowColumns.push('<a href="' + deleteURL + '"><%= trashHelper.isTrashEnabled(scopeGroupId) ? UnicodeLanguageUtil.get(resourceBundle, "move-to-recycle-bin") : UnicodeFormatter.toString(removeAttachmentIcon) %></a>');
+				rowColumns.push('<a class="delete-attachment" data-cmd="<%= trashHelper.isTrashEnabled(scopeGroupId) ? Constants.MOVE_TO_TRASH : Constants.DELETE %>" data-fileName="' + event.name + '" data-nodeId="<%= wikiPage.getNodeId() %>" data-rowId="' + event.id + '" data-ticketKey="<%= ticket.getKey() %>" data-title="<%= wikiPage.getTitle() %>" href="javascript:;"><%= trashHelper.isTrashEnabled(scopeGroupId) ? UnicodeLanguageUtil.get(resourceBundle, "move-to-recycle-bin") : UnicodeFormatter.toString(removeAttachmentIcon) %></a>');
```

Comment:
Instead of passing a bunch or parameters and then create the portletURL in the javascript side, isn't it faster to just create the url here and pass it as a single data-url parameter? Doing this we avoid having to do things such as hardcoding and I think that we don't need the ticket key parameter either, which simplifies things:
`deleteUrl.setPortletId('com_liferay_wiki_web_portlet_WikiPortlet');`

Reply:
👍